udata/migrations/2024-10-01-remove-ods-duplicates.py

```diff
@@ -0,0 +1,89 @@
+"""
+Remove duplicate OpenDataSoft resources
+The duplicate are due to ODS modifying the URL of their CSV and XLSX exports,
+the URL being the identifier of the resources in DCAT harvesting.
+The default /export/csv had been existing since harvesting ODS with DCAT.
+2024-08-07 : `use_labels=false` appended at the end of the URL -> first duplicates
+2024-08-09 : replaced by `use_labels=true` -> second set of duplicates
+"""
+
+import logging
+import traceback
+
+from mongoengine.errors import ValidationError
+
+from udata.models import Dataset
+
+log = logging.getLogger(__name__)
+
+
+def migrate(db):
+    count = 0
+    errors = 0
+
+    # Datasets with use_labels=false are the one that were most surely impacted by the URL modifications
+    datasets = Dataset.objects(resources__url__contains="use_labels=false")
+
+    log.info("Starting")
+    log.info(f"{datasets.count()} datasets to process...")
+    for dat in datasets:
+        for res_format in ["csv", "xlsx"]:
+            resources = [res for res in dat.resources if f"/exports/{res_format}" in res.url]
+            if not resources:
+                # This dataset doesn't have csv or xlsx resources
+                continue
+            if len(resources) not in [3, 4]:
+                log.info(
+                    f"Skipping, {len(resources)} {res_format} duplicate resources found for {dat.id}. We're expecting 3 or 4"
```

Comment:
Indeed! Some resources have old duplicates that were created at migration time I guess, ex [this dataset](https://www.data.gouv.fr/fr/datasets/6350c81620af2751ce90f68e/). I didn't want to dive into these, but since I'm removing the CSV and XLSX duplicates, I remove them if it's the case.

Printing logs allow us to deal with special cases manually after, ex:
* unexpected number of duplicates, ex [here](https://www.data.gouv.fr/fr/datasets/66b40c68c73c0b6bcfa06f1f/). This dataset was indeed created on 08/08/24, right on time for the `use_labels=false`
* unexpected `use_labels=true` all around [here](https://www.data.gouv.fr/fr/datasets/65c810957f6784bf33e5f9fe/)